stdlib/join/join_test.flux

```diff
@@ -442,3 +443,153 @@ testcase join_empty_table {
+                {_time: 2022-09-28T14:18:25Z, _value: "complete", id: "3tagx5d"},
+                {_time: 2022-09-28T14:18:35Z, _value: "complete", id: "3tagx5x"},
+                {_time: 2022-09-28T14:18:37Z, _value: "complete", id: "3tagx6j"},
+                {_time: 2022-10-05T09:15:09Z, _value: "print done", id: "3ucxcj0"},
+                {_time: 2022-10-05T07:06:07Z, _value: "print done", id: "3ucxckb"},
+                {_time: 2022-10-03T08:08:53Z, _value: "received", id: "3vmv47d"},
+                {_time: 2022-10-03T08:08:38Z, _value: "received", id: "3vmyxd6"},
+                {_time: 2022-10-05T07:09:44Z, _value: "print done", id: "3wbtnn8"},
+                {_time: 2022-10-05T07:06:08Z, _value: "print done", id: "3wbtnnk"},
+                {_time: 2022-10-05T09:14:32Z, _value: "print done", id: "3wbtnqd"},
+                {_time: 2022-09-29T13:12:37Z, _value: "print done", id: "3wbtnt0"},
+                {_time: 2022-09-29T15:54:15Z, _value: "on hold", id: "3whmdwb"},
+                {_time: 2022-09-30T14:43:29Z, _value: "Closed", id: "3wx14kx"},
+                {_time: 2022-09-29T13:54:20Z, _value: "finish parts", id: "3wx1ug0"},
+                {_time: 2022-09-29T15:53:59Z, _value: "on hold", id: "3x8apx6"},
+            ],
+        )
+            |> debug.opaque()
+
+    mappedRight =
+        rTbl
+            |> set(key: "on", value: "x")
+
+    mappedLeft =
+        lTbl
+            |> set(key: "on", value: "x")
+    want =
+        array.from(
+            rows: [
+                {_time: 2022-09-28T12:24:30Z, bool_value: true, id: "2cbuegt", on: "x"},
+                {_time: 2022-09-28T17:00:00Z, bool_value: false, id: "2cbuegt", on: "x"},
+                {_time: 2022-09-29T07:00:00Z, bool_value: true, id: "2cbuegt", on: "x"},
+                {_time: 2022-09-29T17:00:00Z, bool_value: false, id: "2cbuegt", on: "x"},
+                {_time: 2022-09-30T08:00:00Z, bool_value: true, id: "2cbuegt", on: "x"},
+                {_time: 2022-09-30T16:00:00Z, bool_value: false, id: "2cbuegt", on: "x"},
+                {_time: 2022-10-03T07:00:00Z, bool_value: true, id: "2cbuegt", on: "x"},
+                {_time: 2022-10-03T17:00:00Z, bool_value: false, id: "2cbuegt", on: "x"},
+                {_time: 2022-10-04T07:00:00Z, bool_value: true, id: "2cbuegt", on: "x"},
+                {_time: 2022-10-04T17:00:00Z, bool_value: false, id: "2cbuegt", on: "x"},
+                {_time: 2022-10-05T07:00:00Z, bool_value: true, id: "2cbuegt", on: "x"},
+                {_time: 2022-10-05T12:24:30Z, bool_value: true, id: "2cbuegt", on: "x"},
+                {_time: 2022-10-05T12:24:30Z, bool_value: true, id: "2cbuegt", on: "x"},
+            ],
+        )
+
+    got =
+        join.inner(
+            left: mappedLeft,
+            right: mappedRight,
+            on: (l, r) => l.on == r.on,
+            as: (l, r) => {
+                return {r with id: l.id}
+            },
+        )
+            |> limit(n: 13)
```

Comment:
Since you limit the number of rows here, it's hard to tell if there is an off-by-one error in the fix. Maybe verify the row count as well?

Reply:
I did limit it so that I do not have to add a lot of rows in the `want`. 
I wrote it in a different way now, as each `id` from the left table will join with 13 rows of the right table producing 13 rows per id for the join result, I have added `tail(n:13)` to verify their is no off-by-one error.